python/cudf/cudf_pandas_tests/test_cudf_pandas.py

```diff
@@ -1421,3 +1421,18 @@ def test_holidays_within_dates(holiday, start, expected):
             utc.localize(xpd.Timestamp(start)),
         )
     ) == [utc.localize(dt) for dt in expected]
+
+
+def test_pandas_debugging_mode_option(monkeypatch):
+    from cudf import Series, set_option
+
+    set_option("mode.pandas_debugging", True)
+
+    def mock_mean(self, *args, **kwargs):
+        return 1.0
+
+    monkeypatch.setattr(Series, "mean", mock_mean)
+    with pytest.warns(UserWarning):
+        s = xpd.Series([1, 2])
+        assert s.mean() == 1.0
+    set_option("mode.pandas_debugging", False)
```

Comment:
Use an [option context manager](https://github.com/rapidsai/cudf/blob/ff981a4048a389b0e2582e94d3397a83096d16c9/python/cudf/cudf/options.py#L312-L324) instead of setting and resetting the global options. Also make sure the warning has the right message.

```suggestion
    from cudf import Series, option_context

    def mock_mean(self, *args, **kwargs):
        return 1.0

    with option_context("mode.pandas_debugging", True)
        monkeypatch.setattr(Series, "mean", mock_mean)
        s = xpd.Series([1, 2])
        with pytest.warns(UserWarning, match="The results from cudf and pandas were different."):
            assert s.mean() == 1.0
```